### ISSUE TITLE
Faulty properties section in the extension package.json file breaks accessing configuration system wide.

### ISSUE THREAD

**Issue by oleg-shilo** [Microsoft/vscode#49530]
- VSCode Version: 1.23.0
- OS Version: Win10

Steps to Reproduce:

1. Have a miss-configured property in the package.json:
  ```json
  ...
  "contributes": {
    "configuration": {
      "type": "object",
      "title": "HelloWorld configuration",
      "properties": {
        "HelloWorld.PropA": {
          "type": "string",
          "default": "test_value"
        },
        "HelloWorld.PropB": "invalid_test_value"
      }
    }
  ``` 
2. Start debugging extension.
3. Observe multiple extensions failing to initialize in the '[Extension Development Host]':
  ![image](https://user-images.githubusercontent.com/16729806/39813059-54eac068-53d2-11e8-97a5-06f4874cbe59.png)
4. Stop debugging and remove `"HelloWorld.PropB": "invalid_test_value"` line from the package.json.
5. Start debugging extension again.
6. Observe all extensions working as expected

Does this issue occur when all extensions are disabled?: No

----

The problem is that the offending extension affects all extensions **system wide**. Thus any extension that reads its own config will fail to do so. If such an extension does not handle exceptions then it will fail to initialize completely.

The problem is quite severe. It affects the all core extensions (e.g. TypeScript, CSharp). Meaning that one bad extension from MarketPlace can cripple the whole IDE.
 
I have created the test-case repo which demonstrates the problem: https://github.com/oleg-shilo/vscode-issue-reporting.git

**Comment by sandy081** [comment]
@oleg-shilo I am not able to reproduce it. I used the repo your provided and on VS Code 1.23.0. Followed the same steps as you described.

Let me know if there is anything missing

**Comment by oleg-shilo** [comment]
In the initial test that I conducted I started with a clean VSCode installation with no extensions. Then I added code-spellchecker and that repo that I shared. 

I had suspicion that the affected extensions are the one that are initialized after the offending (faulty) extension. Thus I decided to go with a clean environment for testing.

Just before reporting the problem (after I installed my all usual extension) I repeated the test again and it was the same outcome. 

I'll try to repeat it on another PC tomorrow. Will let you know.

**Comment by sandy081** [comment]
I see.. will also try the same.

**Comment by oleg-shilo** [comment]
Tested on another PC. The same result:

![image](https://user-images.githubusercontent.com/16729806/40092138-14289000-58ff-11e8-8ace-77336311cc08.png)

![image](https://user-images.githubusercontent.com/16729806/40092132-057364f4-58ff-11e8-980c-f07484d5f948.png)

Just in case I have attached the very same project that upsets the IDE in my test.
[vscode-issue-reporting.zip](https://github.com/Microsoft/vscode/files/2007235/vscode-issue-reporting.zip)

**Comment by sandy081** [comment]
This if fixed by #49339
